components/tools/OmeroPy/test/integration/gatewaytest/test_image.py

```diff
@@ -367,3 +371,6 @@ def testExport(self, gatewaywrapper):
         finally:
             gatewaywrapper.loginAsAdmin()
             admin = gatewaywrapper.gateway.getAdminService()
+        # Check that the exporter is closed
+        for v in gatewaywrapper.gateway.c.getSession().activeServices():
+            assert 'ExporterPrx' not in v, 'Leaked exporter!'
```

Comment:
I removed the ```close()``` calls in BlitzGateway and this test still passed.
When I added ```print v``` to this loop, I saw this
```
5b494fb5-be0d-4af2-980b-e9d3d0066dc8omero.api.Exporter
```
which doesn't contain ```ExporterPrx```

Reply:
pushing a fix i.e. Exporter instead of ExporterPrx

Reply:
I think you also need to test for open services after the ```del gen``` above. Any services left open at that point are not picked up by by the checking at the end of the test since by that time we have logged in as a different user.